client/player/dashjsVmafTV/index.html

```diff
@@ -0,0 +1,77 @@
+    <meta name="viewport" content="width=device-width, initial-scale=1.0">
+    <script type="text/javascript" src="http://localhost/player/dashjsVmafTV/asset/4.3.0.debug.dash.js"></script>
+    <script src="https://cdn.jsdelivr.net/npm/axios/dist/axios.min.js"></script>
+</head>
+<body>
+<div>
+    <video style="width: 1280px;height: 720px;" id="video"></video>
+</div>
+
+<div id='trace' style="height:640px;width:480px;border:1px solid #ccc;font:16px/26px Georgia, Garamond;overflow:auto;"></div>
+
+<script type="text/javascript">
+    // source https://streaminglearningcenter.com/encoding/best-practices-for-netflixs-vmaf-metric.html
+    // source https://eyevinntechnology.medium.com/keep-an-eye-on-the-video-quality-b9bcb58dd5a1
+    const minAcceptableVMAF = 20;
+    const maxAcceptableVMAF = 93;
+    
+    const context = this.context
+  document.addEventListener('DOMContentLoaded', () => {
+    const keyValues = window.location.search.substr(1).split('&')
+    let experimentId, title, alk, index
+    
+    keyValues.forEach(keyValueString => {
+      let keyValueArray = keyValueString.split('=')
+      switch (keyValueArray[0]) {
+        case 'id':
+          experimentId = keyValueArray[1]
+          break
+        case 'title':
+          title = keyValueArray[1]
+          break
+        case 'alk':
+          alk = keyValueArray[1]
+          break
+        case 'index':
+          index = keyValueArray[1]
+          break
+      }
+    })
+    const mpdURL = 'http://localhost/dataset/' + title + '/manifest.mpd?playerABR=dashjsVmafTV-dynamic-' + index
+    let logURL = 'http://localhost/log/' + title + '/%event%?playerABR=dashjsVmafTV-dynamic-' + index
+
+    const video = document.getElementById('video')
+    const player = dashjs.MediaPlayer().create()
+
+    // const events = ["loadstart",  "suspend", "abort", "error", "emptied", "stalled", "loadedmetadata", "loadeddata", "canplay", "canplaythrough", "playing", "waiting", "seeking", "seeked", "ended", "durationchange", "play", "pause", "ratechange", "resize", "volumechange", "timeupdate", "progress"];
+    const events = ['loadstart', 'suspend', 'abort', 'error', 'emptied', 'stalled', 'loadedmetadata', 'loadeddata', 'canplay', 'canplaythrough', 'playing', 'waiting', 'seeking', 'seeked', 'ended', 'durationchange', 'play', 'pause', 'ratechange']
+
+    events.forEach(event => {
+      video.addEventListener(event, e => {
+        axios.get(logURL.replace('%event%', e.type) + '&noCache=' + (new Date()).getTime()).catch(error => {
+          console.log(error)
+        })
+      })
+    })
+    
+    // limit bitrate based on VMAF
+    var filterCapabilities = function (representation) {
+        if (representation.mimeType === "video/mp4" && representation.vmaf > 0) {
+            return representation.vmaf >= minAcceptableVMAF && representation.vmaf <= maxAcceptableVMAF;
+        }
+        return true;
+    }
+    player.registerCustomCapabilitiesFilter(filterCapabilities) ;
```

Comment:
space before the semicolumn might cause problems in some situations

Reply:
Updated the code.